From: Tias Guns <tias@ulyssis.org>
Date: Tue, 20 Mar 2012 23:03:44 +0000
Subject: [PATCH] fix readahead, add syscall

fix readahead, add simple syscall
Can only be tested with LFS support (not patched yet)

from 'misc-syscalls' by Dan Drown
http://dan.drown.org/android/src/busybox/
---
 libbb/Kbuild.src  |    1 +
 libbb/readahead.c |    7 +++++++
 2 files changed, 8 insertions(+), 0 deletions(-)
 create mode 100644 libbb/readahead.c

diff --git a/libbb/Kbuild.src b/libbb/Kbuild.src
index 1cb1427..97c2738 100644
--- a/libbb/Kbuild.src
+++ b/libbb/Kbuild.src
@@ -201,6 +201,7 @@ lib-$(CONFIG_HOSTNAME) += sethostname.o
 
 lib-$(CONFIG_ADJTIMEX) += adjtimex.o
 lib-$(CONFIG_SWAPONOFF) += swapon.o swapoff.o
+lib-$(CONFIG_READAHEAD) += readahead.o
 
 lib-$(CONFIG_SYSLOGD) += semop.o shmdt-at.o
 
diff --git a/libbb/readahead.c b/libbb/readahead.c
new file mode 100644
index 0000000..c0474f9
--- /dev/null
+++ b/libbb/readahead.c
@@ -0,0 +1,7 @@
+#include <fcntl.h>
+#include <sys/syscall.h>
+#include <stdio.h>
+
+ssize_t readahead(int fd, off64_t offset, size_t count) {
+  return syscall(__NR_readahead, fd, offset, count);
+}
-- 
1.7.0.4

